pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -667,7 +667,14 @@ func parseOneSystemAdapterConfig(getconfigCtx *getconfigContext,
 	} else {
 		port.Phylabel = phyio.Phylabel
 		port.Logicallabel = phyio.Logicallabel
-		port.IfName = phyio.Phyaddr.Ifname
+		if phyio.Phyaddr.Ifname == "" {
+			// Might not be set for all models
+			port.IfName = phyio.Logicallabel
```

Comment:
Did you mean to set it to Phylabel here?
What if LogicalLabel is also not set?
So the algo:

if phyio.Phyaddr.Ifname != "" {
   port.IfName = phyio.Phyaddr.Ifname
} else if phyio.Logicallabel != "" {
    port.IfName = phyio.Logicallabel
} else {
    port.IfName = phyio.Physicallabel
}

Reply:
Thinking more.. Should we also add this comment in the Proto file? Or leave it to the discretion of EVE? Essentially enshrine it in the API that the above is the expectted behavior?

phylabel is primary key - so we can assume it is always present..

Reply:
Go comments. Updated code.
Also, I realized we should make sure the AssignableAdapter IoBundle has ifname set for network adapters, since we do comparisons against that.